docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+
+#### new UserContacts-Table
+
+A new entity `UserContacts `is introduced to store several contacts of different types like email, telephone or other kinds of contact addresses.
+
+| Column          | Type   | Description                                                                                                                                                            |
+| --------------- | ------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| id              | int    | the technical key of a contact entity                                                                                                                                  |
+| type            | int    | Defines the type of contact entry as enum: Email, Phone, etc                                                                                                           |
+| userID          | int    | Defines the foreign key to the `Users` table                                                                                                                         |
+| email           | String | defines the address of a contact entry of type Email                                                                                                                   |
+| phone           | String | defines the address of a contact entry of type Phone                                                                                                                   |
+| contactChannels | String | define the contact channel as comma separated list for which this entry is confirmed by the user e.g. main contact (default), infomail, contracting, advertisings, ... |
+
+### Database-Migration
+
+After the adaption of the database schema and to keep valid consistent data, there must be several steps of data migration to initialize the new and changed columns and tables.
+
+#### Initialize GradidoID
+
+In a one-time migration create for each entry of the `Users `tabel an unique UUID (version4).
+
+#### Primary Email Contact
+
+In a one-time migration read for each entry of the `Users `table the `Users.id` and `Users.email` and create for it a new entry in the `UsersContact `table, by initializing the contact-values with:
+
+* id = new technical key
+* type = Enum-Email
+* userID = `Users.id`
+* email = `Users.email`
+* phone = null
+* usedChannel = Enum-"main contact"
+
+and update the `Users `entry with `Users.emailId = UsersContact.Id` and `Users.passphraseEncryptionType = 1`
+
+After this one-time migration the column `Users.email` can be deleted.
+
+### Adaption of BusinessLogic
+
+The following logic or business processes has to be adapted for introducing the Gradido-ID
+
+#### Read-Write Access of Users-Table especially Email
+
+The ORM mapping has to be adapted to the changed and new database schema.
```

Comment:
This is actually part of the migration implementation. New entity models are defined alongside the migration

Reply:
Ok if the migration will use the ORM logic than it must be adapted before.
If the migration will be done directly on database per sql scripts, it will be indepent from ORM.

Feel free how to implement the migration.